src/needs-parens.js

```diff
@@ -161,43 +164,13 @@ function needsParens(path) {
         return true;
       }
 
-      return false;
-    case "propertylookup":
-    case "staticlookup":
-    case "offsetlookup":
       return false;
     case "print":
-      if (parent.kind === "bin") {
-        return true;
-      }
-
-      return false;
-    case "bin": {
-      if (["pre", "post"].includes(parent.kind)) {
-        return true;
-      }
-
-      if (["if", "while", "do", "switch", "case"].includes(parent.kind)) {
-        return false;
-      }
-
-      // $var = false or true;
-      // The constant false is assigned to $f before the "or" operation occurs
-      // Acts like: (($var = false) or true)
-      if (
-        node.right.kind === "bin" &&
-        ["and", "xor", "or"].includes(node.right.type)
-      ) {
-        return true;
-      }
-
-      if (parent.kind === "cast") {
-        return true;
-      }
-    }
+    case "include":
+      return parent.kind === "bin";
   }
 
-  return node.parenthesizedExpression;
+  return false;
```

Comment:
By default we don't print parens, it is allow to catch all problems with parens, `prettier` use same logic https://github.com/prettier/prettier/blob/master/src/language-js/needs-parens.js#L604